pkgs/applications/networking/kubo-migrator/src.nix

```diff
@@ -0,0 +1,42 @@
+{ lib
+, stdenvNoCC
+, fetchFromGitHub
+, fetchpatch
+}:
+
+stdenvNoCC.mkDerivation rec {
+  name = "kubo-migrator-src";
+
+  src = fetchFromGitHub {
+    owner = "ipfs";
+    repo = "fs-repo-migrations";
+    rev = "fs-repo-12-to-13/v1.0.0";
+    hash = "sha256-QQone7E2Be+jVfnrwqQ1Ny4jo6mSDHhaY3ErkNdn2f8=";
+  };
+
+  patches = [
+    # Make the migrations compatible with Go 1.17 and newer
+    (fetchpatch {
+      name = "fix-fs-repo-10-to-11.patch";
+      url = "https://github.com/ipfs/fs-repo-migrations/pull/163/commits/c3b39a3ccd51e65e1039652235cb6930d3304a08.patch";
+      hash = "sha256-YkFtFsYueCd24WsRINq7Q5wvFjoIVAqlN9yy9y6kNrw=";
+    })
+    (fetchpatch {
+      name = "fix-fs-repo-11-to-12.patch";
+      url = "https://github.com/ipfs/fs-repo-migrations/pull/163/commits/408f58ffdf9869bcb51cd4d8d99d5aefd65df593.patch";
+      hash = "sha256-IkRkwNVDLSU8wWmtFuxM5H0g6Jkkso/i8k5XmjMrLA8=";
+    })
+  ];
```

Comment:
i suggest to commit this as one patchfile to nixpkgs

Reply:
Why? Your PR is not going to disappear. How would committing the patch to Nixpkgs be better?

Reply:
generally, we should avoid dependency on github
the source can be hosted on other git providers, and then the patch is missing

Reply:
I have archived the patches: https://web.archive.org/web/20230125112511/https://github.com/ipfs/fs-repo-migrations/commit/c3b39a3ccd51e65e1039652235cb6930d3304a08.patch https://web.archive.org/web/20230125112541/https://github.com/ipfs/fs-repo-migrations/commit/408f58ffdf9869bcb51cd4d8d99d5aefd65df593.patch. If the source ever disappears for any reason, we can still access the patches from archive.org.
I try to avoid adding patches to Nixpkgs directly unless necessary to keep the size down. I think fetching a patch like this is perfectly acceptable in this case.